backend/lcfs/web/api/fuel_code/repo.py

```diff
@@ -189,12 +194,10 @@ async def get_transport_mode(self, transport_mode_id: int) -> TransportMode:
         )
 
     @repo_handler
-    async def get_transport_mode_by_name(self, mode_name: str) -> TransportMode:
-        query = select(TransportMode).where(TransportMode.transport_mode == mode_name)
-        result = await self.db.execute(query)
-        transport_mode = await result.scalar_one_or_none()
-
-        return transport_mode
+    async def get_transport_mode_by_name(self, transport_mode: str) -> TransportMode:
```

Comment:
should add a check for when no matching TransportMode records are returned.

Reply:
Changes from scalar_one_or_none to scalar_one so it throws an exception.